index.js

```diff
@@ -139,12 +142,65 @@ module.exports = {
     return fastbootBuild.toTree();
   },
 
+  serverMiddleware: function(options) {
+    var emberCliVersion = this._getEmberCliVersion();
+    var app = options.app;
+    var options = options.options;
+
+    if (emberCliVersion.satisfies('>= 2.12.0-beta.1')) {
+      // only run the middleware when ember-cli version for app is above 2.12.0-beta.1 since
+      // that version contains API to hook fastboot into ember-cli
+
+      app.use(function(req, resp, next) {
+        var broccoliHeader = req.headers['x-broccoli'];
+        var outputPath = broccoliHeader['outputPath'];
+
+        if (broccoliHeader['url'] === req.serveUrl) {
+          // if it is a base page request, then have fastboot serve the base page
+          if (!this.fastboot) {
+            // TODO: make this configurable for allowing apps to pass sandboxGlobals
+            // and custom sandbox class
+            this.ui.writeLine(chalk.green('App is being served by FastBoot'));
+            this.fastboot = new FastBoot({
+              distPath: outputPath
+            });
+          }
+
+          var fastbootMiddleware = FastBootExpressMiddleware({
+            fastboot: this.fastboot
+          });
+
+          fastbootMiddleware(req, resp, next);
```

Comment:
Yes that is correct. If I understand correctly, FastBoot is only meant for initial requests and that too for the base page request (basically have your app be server side rendered for the initial request). For all other requests (other assets, subsequent transitions) should be defaulted to `ember-cli`'s asset serving. 

Do you think that is a wrong assumption? Is there a usecase that I might have missed?